pkg/core/native/native_neo.go

```diff
@@ -810,20 +825,23 @@ func (n *NEO) VoteInternal(ic *interop.Context, h util.Uint160, pub *keys.Public
 	if err := n.distributeGas(ic, h, acc); err != nil {
 		return err
 	}
-	if err := n.ModifyAccountVotes(acc, ic.DAO, new(big.Int).Neg(&acc.Balance), false); err != nil {
+	// TODO: here we modify cached validators value two times in a raw. Between
+	//  these changes we have invalid cached validators state, so maybe it's
+	//  better to update cache.validators once after votes modification is completed.
```

Comment:
Let `modifyAccountVotes` handles votes, it also makes little sense recalculating validators with every voting NEO transfer since the updated value is only needed once in a committee cycle.

Reply:
First note is fixed, the second one isn't.

But how about other users of `validators` field? Before we had `nil`-ablle validators field, and in case if someone needed it, we could always recalculate it from the storage. Now the caching scheme is different, and I'm not sure we can update validators once in a while (not after each votes modification).

Reply:
We only update committee in `OnPersist` and only if `ShouldUpdateCommitteeAt(ic.Block.Index)`. `getCommittee` never touches storage, it operates cache-only even in the current master branch. Same thing with `getNextBlockValidators`. So why should we do anything on every vote if we've never done that? This might be related to your problem with consensus test failure.

Reply:
I wasn't able to make those commits work properly and dropped two commits related to the committee and validators cache refactoring. Validators cache works fine and we doesn't need to change it, validators are updated exactly when needed. But I'm still not sure about committee related cache. At least now the test passes.